src/testing/cluster/message_bus.zig

```diff
@@ -70,6 +72,15 @@ pub const MessageBus = struct {
         bus.pool.unref(message);
     }
 
+    pub fn resume_needed(bus: *MessageBus) bool {
+        return bus.suspended;
+    }
+
+    pub fn resume_receive(bus: *MessageBus) void {
```

Comment:
Can we assert these transitions?

Reply:
I don't think so? The contract is that you can call `.resume_receive` whenever, there isn't a wrong time to call it. And in the callback we do assert that the resume was requested. That's on the level of message bus. 

On the level of replica it is tricky. We do want to enforce that, if we suspend a message, we eventually call `resume_receive`, to ensure we don't deadlock. I don't see a super great way to solve that, but my assert in tick did in fact caught a bug where I was missing a resumption call. 

Or am I misunderstanding the kind of assertions you have in mind?